Source/Tests/LinuxSyscalls/Syscalls.cpp

```diff
@@ -226,89 +212,156 @@ static bool IsSupportedByInterpreter(std::string const &Filename) {
+
+    if (!IsShebangFilename(Filename) && Type == ELFLoader::ELFContainer::ELFType::TYPE_NONE) {
+      // If our interpeter doesn't support this file format AND ELF format is NONE then ENOEXEC
+      // binfmt_misc could end up handling this case but we can't know that without parsing binfmt_misc ourselves
+      // Return -ENOEXEC until proven otherwise
+      return -ENOEXEC;
+    }
   }
 
   // If we don't have the interpreter installed we need to be extra careful for ENOEXEC
   // Reasoning is that if we try executing a file from FEXLoader then this process loses the ENOEXEC flag
   // Kernel does its own checks for file format support for this
   // We can only call execve directly if we both have an interpreter installed AND were ran with the interpreter
   // If the user ran FEX through FEXLoader then we must go down the emulated path
-  ELFLoader::ELFContainer::ELFType Type = ELFLoader::ELFContainer::GetELFType(Filename);
   uint64_t Result{};
   if (FEX::HLE::_SyscallHandler->IsInterpreterInstalled() &&
       FEX::HLE::_SyscallHandler->IsInterpreter() &&
       (Type == ELFLoader::ELFContainer::ELFType::TYPE_X86_32 ||
        Type == ELFLoader::ELFContainer::ELFType::TYPE_X86_64)) {
     // If the FEX interpreter is installed then just execve the ELF file
     // This will stay inside of our emulated environment since binfmt_misc will capture it
-    if (Args) {
-      Result = ::syscall(SYS_execveat, Args->dirfd, Filename.c_str(), argv, envp, Args->flags);
-    }
-    else {
-      Result = execve(Filename.c_str(), argv, envp);
-    }
+    Result = ::syscall(SYS_execveat, Args.dirfd, Filename.c_str(), argv, envp, Args.flags);
     SYSCALL_ERRNO();
   }
 
-  if (!IsSupportedByInterpreter(Filename) && Type == ELFLoader::ELFContainer::ELFType::TYPE_NONE) {
-    // If our interpeter doesn't support this file format AND ELF format is NONE then ENOEXEC
-    // binfmt_misc could end up handling this case but we can't know that without parsing binfmt_misc ourselves
-    // Return -ENOEXEC until proven otherwise
-    return -ENOEXEC;
-  }
-
   if (Type == ELFLoader::ELFContainer::ELFType::TYPE_OTHER_ELF) {
     // We are trying to execute an ELF of a different architecture
     // We can't know if we can support this without architecture specific checks and binfmt_misc parsing
     // Just execve it and let the kernel handle the process
-    if (Args) {
-      Result = ::syscall(SYS_execveat, Args->dirfd, Filename.c_str(), argv, envp, Args->flags);
-    }
-    else {
-      Result = execve(Filename.c_str(), argv, envp);
-    }
+    Result = ::syscall(SYS_execveat, Args.dirfd, Filename.c_str(), argv, envp, Args.flags);
     SYSCALL_ERRNO();
   }
 
   // We don't have an interpreter installed or we are executing a non-ELF executable
   // We now need to munge the arguments
   std::vector<const char *> ExecveArgs{};
+  std::vector<const char *> EnvpArgs{};
+  char *const *EnvpPtr = envp;
+  const char NullString[] = "\0";
```

Comment:
Is it important to use `"\0"` over simply `""` here?

Reply:
Nope, changed.